lib/_http_server.js

```diff
@@ -530,14 +530,14 @@ function onParserExecuteCommon(server, socket, parser, state, ret, d) {
     parser = null;
 
     var eventName = req.method === 'CONNECT' ? 'connect' : 'upgrade';
-    if (server.listenerCount(eventName) > 0) {
+    if (eventName === 'upgrade' || server.listenerCount(eventName) > 0) {
```

Comment:
can you combine this if with the ternary before somehow? With this change we truly only need to do `server.listenerCount('connect')` here.

Reply:
I don't think we can combine it with the ternary since the `eventName` is also used inside the if clause when emitting. We could change `server.listenerCount(eventName)` to be `server.listenerCount('connect')` but I'm not sure that's truly better...

Reply:
As I said, I'm not convinced. Removing the ternary might create code that is harder to read in the end.

Reply:
@mcollina ... are you ok with this landing without this change?

Reply:
To be clear, I would be happy to make a change but I'm not sure what it would be. I don't think there's a way to simplify this since `eventName` is used inside the body of the if statement.

Reply:
The only thing I can think of is reversing the condition but it doesn't change anything.

```js
if (eventName === 'connect' && server.listenerCount('connect') === 0) {
  socket.destroy();
} else {
  // ...
}
```